src/Ast.ml

```diff
@@ -1628,6 +1628,10 @@ end = struct
       | Pexp_apply (({pexp_desc= Pexp_new _; _} as exp2), _)
         when exp2 == exp ->
           false
+      | Pexp_apply
+          ({pexp_desc= Pexp_ident {txt= Lident txt; _}}, (_, first) :: _)
+        when String.is_prefix ~prefix:"#" txt && (not (first == exp)) -> (
+        match exp.pexp_desc with Pexp_ident _ -> false | _ -> true )
```

Comment:
Ok, so the hash infix ops have a different precedence than the others. I think it would be better to fix this by adding `&& Poly.(prec_ast ctx < Some Apply)` to the `when` guard in `is_right_infix_arg` below.